tests/test_curate_preferences.py

```
@@ -297,5 +438,213 @@ def test_invalid_utf8_fails_closed(self):
+
+    def test_fixture_sources_stay_byte_identical_and_writer_refuses_unsafe_paths(self):
+        with tempfile.TemporaryDirectory() as td:
+            destination = Path(td)
+            run = curate_preferences.curate_source(PURITY_FIXTURES)
+            curate_preferences.write_run(
+                run,
+                PURITY_FIXTURES,
+                destination / "preferences.jsonl",
+                destination / "manifest.jsonl",
+            )
+            with self.assertRaisesRegex(
```

Ruff (0.16.1)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)